src/app/ixy-pcap.c

```diff
@@ -0,0 +1,86 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/time.h>
+
+#include "driver/device.h"
+
+const int BATCH_SIZE = 32;
+
+// From https://wiki.wireshark.org/Development/LibpcapFileFormat
+typedef struct pcap_hdr_s {
+	uint32_t magic_number;  /* magic number */
+	uint16_t version_major; /* major version number */
+	uint16_t version_minor; /* minor version number */
+	int32_t  thiszone;      /* GMT to local correction */
+	uint32_t sigfigs;       /* accuracy of timestamps */
+	uint32_t snaplen;       /* max length of captured packets, in octets */
+	uint32_t network;       /* data link type */
+} pcap_hdr_t;
+
+typedef struct pcaprec_hdr_s {
+	uint32_t ts_sec;        /* timestamp seconds */
+	uint32_t ts_usec;       /* timestamp microseconds */
+	uint32_t incl_len;      /* number of octets of packet saved in file */
+	uint32_t orig_len;      /* actual length of packet */
+} pcaprec_hdr_t;
+
+int main(int argc, char* argv[]) {
+	if (argc < 3 || argc > 4) {
+		printf("Usage: %s <pci bus id> <output file> [n packets]\n", argv[0]);
+		return 1;
+	}
+
+	struct ixy_device* dev = ixy_init(argv[1], 1, 1, 0);
+
+	FILE* pcap = fopen(argv[2], "wb");
+	if (pcap == NULL) {
+		error("failed to open file %s", argv[2]);
+	}
+
+	int n_packets = -1;
```

Comment:
can you change that to uint64_t? 2 million packets isn't a lot

Reply:
Changed it to int64_t to account for unbounded capture. (Though it doesn't really make a difference in practice since -1 would just be just be UINT64_MAX.)